test/system/assignments_test.rb

```diff
@@ -2,6 +2,8 @@
 
 class AssignmentsTest < ApplicationSystemTestCase
   setup do
+    Client.with_deleted.map(&:really_destroy!)
+    Volunteer.with_deleted.map(&:really_destroy!)
```

Comment:
I guess this can be removed after you added the other workaround in `test_helper.rb` ;-)